### DISCUSSION TITLE
Nested <svg> elements - question and feature request

### DISCUSSION THREAD

**Discussion by idw-git** [AngusJohnson/Image32#115] [General]
Angus, as far as I can see your SVG parser doesn't seem to be capable of parsing nested <svg> elements, instead it simply appears to abort further processing. Is this correct and are their any plans to add this feature? I can provide a pretty simple test sample if required.

**Comment by AngusJohnson** [comment]
> Angus, as far as I can see your SVG parser doesn't seem to be capable of parsing nested elements

Virtually all elements in SVG files are nested, so of course it parses nested elements.

**Reply by idw-git** [reply]
Sorry, my fault, I should have been more specific (I think I didn't realize that the final version of my initial post clipped the second **svg** element name due to using unencoded angle brackets): I'm referring to &lt;svg&gt; elements  being nested within &lt;svg&gt; elements,  zipped sample attached. [Nested.zip](https://github.com/user-attachments/files/18000644/Nested.zip)

For comparison here's also a screenshot of the Firefox rendering (left) versus the one using your sample Svg101.exe.

![Screenshot](https://github.com/user-attachments/assets/8c6afa9c-251d-45f6-b974-ac74e776f678)

**Comment by AngusJohnson** [comment]
Actually Image32 does parse the nested SVG element, but it's currently ignoring its `x` and `y` attributes, so the nested SVG image in your example above is being drawn on top of the elements drawn in the outer SVG element.

Edit: Should be fixed now.

**Reply by idw-git** [reply]
Thanks a lot!
